stl/inc/numbers

```diff
@@ -0,0 +1,98 @@
+// numbers standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _NUMBERS_
+#define _NUMBERS_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20
+#pragma message("The contents of <numbers> are available only with C++20 or later.")
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+#ifdef __cpp_lib_concepts
+#include <concepts>
+#else // ^^^ defined(__cpp_lib_concepts) / !defined(__cpp_lib_concepts) vvv
+#include <type_traits>
+#endif
+
+_STD_BEGIN
+namespace numbers {
+#ifdef __cpp_lib_concepts
+    template <floating_point _Floating>

```

Comment:
> The Standard depicts an unconstrained primary template

Yes it does, and instantiations of the primary template must be ill-formed. WG21 and the proposal authors went around in circles trying to specify the primary template in such a way that it would work with user-defined types, but quickly determined that there was no initializer expression that would correctly initialize *all* possible user-defined types. Had this been a class template we would have declared-but-not-defined the primary template to force specialization, but that's not possible with a variable template so we made instantiation of the primary ill-formed instead.